### ISSUE TITLE
Found an ExtendTemplate node inside a SyntaxTree. Extend nodes must be the root of a tree.

### ISSUE THREAD

**Issue by drbldgtl** [namics/NitroNet#15]
Perhaps I am missing how to do this from the documentation, but I'm trying to render an ASP.NET page with a master layout.

My page view (TempGeneralContentPage) is as follows:
```
{{< TempMaster }}
some body text...
{{ Content.CreatorName }}
```

My master layout (TempMaster) is as follows:
```
<!DOCTYPE html>
<html lang="en">
<head>
    <title>hello!</title>
</head>
<body>
    {{ body }}
</body>
</html>
```

Stack trace excerpt:
```
[VeilCompilerException: Found an ExtendTemplate node inside a SyntaxTree. Extend nodes must be the root of a tree.]
   Veil.VeilEngine.CompileNonGeneric(String templateId, ITemplateParser parser, TextReader templateContents, Type modelType) +677
   NitroNet.ViewEngine.ViewEngines.VeilViewEngine.CreateNonGenericView(String templateId, String content, IVeilEngine viewEngine) +113
   NitroNet.ViewEngine.ViewEngines.<CreateViewAsync>d__4.MoveNext() +807
   System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() +31
   System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) +62
   NitroNet.Mvc.<GetViewResult>d__8.MoveNext() +1429

[AggregateException: One or more errors occurred.]
   System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification) +4719080
   NitroNet.Mvc.NitroNetViewEngine.FindView(ControllerContext controllerContext, String viewName, String masterName, Boolean useCache) +117
   System.Web.Mvc.<>c__DisplayClass6.<FindView>b__4(IViewEngine e) +45
   System.Web.Mvc.ViewEngineCollection.Find(Func`2 lookup, Boolean trackSearchedPaths) +114
   System.Web.Mvc.ViewEngineCollection.FindView(ControllerContext controllerContext, String viewName, String masterName) +220
```

As you can see I'm using NitroNet, and I'm not sure from the stack trace whether this is a misunderstanding on my part, or a bug in Veil or NitroNet. I have cross-posted this issue in the Veil project as well - https://github.com/csainty/Veil/issues/23.

Could you help please? This is my last blocker in getting the engine to work with Umbraco :)

**Comment by csainty** [comment]
This exception should mean the reference to the master page was not the first node in the template.
Is there any whitespace coming before it?

**Comment by drbldgtl** [comment]
No, there is no whitespace before the reference to the master page - the templates are exactly as I entered them above.

**Comment by csainty** [comment]
This might start to fall in to the extensions made to Veil in this project then. 
I'll probably look at Veil a little in the coming weeks for Core 2.0 stuff, so I can try these templates out then and make sure the tests are covering this case.

**Comment by hombreDelPez** [comment]
Thank you for registering this issue.

We have investigated the issue and it turned out that with our modified version of Veil it is not possible to use the master page feature. This is because we made some major changes to the class *VeilEngine* and thus we are not passing the *VeilContext* here.

One of our big goals respectively task in the near future is to make the underlying handlebars parser updatable (i.e. the parser stays unmodified). To achieve this we will have to get in contact with Chris Sainty and we will have to see if and how we possibly can transfer some of our changes back to Veil and how we can make possible modifications of our own work with the parser.

Concerning your issue, a possible way to mitigate the absence of master templates is to extract all the common parts into partials or components [(e.g. head element)](https://github.com/namics/NitroNet/blob/master/docs/samples.md#components-1). If you want to use partials you need to keep in mind to use the "with" expression to [pass the right context](https://github.com/namics/NitroNet/blob/master/docs/known-issues.md#partials-with-passed-context).

We are sorry for this missing feature and hope that the suggested workaround suits your needs.
